static/app/utils/useLocation.tsx

```diff
@@ -0,0 +1,11 @@
+import {useContext} from 'react';
+
+import {RouteContext} from 'sentry/views/routeContext';
+
+export function useLocation() {
+  const route = useContext(RouteContext);
+  if (route === null) {
+    throw new Error('useLocation called outside of routes provider');
+  }
+  return route?.location;
```

Comment:
I think we can expect this to always be present right?
```suggestion
  return route.location;
```